python/postprocessing/modules/common/PrefireCorr.py

```diff
@@ -0,0 +1,123 @@
+import ROOT
+import math
+ROOT.PyConfig.IgnoreCommandLineOptions = True
+import re
+import os
+
+from PhysicsTools.NanoAODTools.postprocessing.framework.datamodel import Collection 
+from PhysicsTools.NanoAODTools.postprocessing.framework.eventloop import Module
+
+class PrefCorr(Module):
+    def __init__(self, jetroot="L1prefiring_jetpt_2017BtoF.root", jetmapname="L1prefiring_jetpt_2017BtoF", photonroot="L1prefiring_photonpt_2017BtoF.root", photonmapname="L1prefiring_photonpt_2017BtoF", UseEMpT=0):
+# UseEMpT: Set to 1 if the jet map is defined for energy deposited in ECAL (pT_EM vs pT). For jet map only, not photon!
+
+        cmssw_base = os.getenv('CMSSW_BASE')
+
+        self.photon_file = self.open_root(cmssw_base + "/src/PhysicsTools/NanoAODTools/data/prefire_maps/" + photonroot)
+        self.photon_map = self.get_root_obj(self.photon_file, photonmapname)
+
+        self.jet_file = self.open_root(cmssw_base + "/src/PhysicsTools/NanoAODTools/data/prefire_maps/" + jetroot)
+        self.jet_map = self.get_root_obj(self.jet_file, jetmapname)
+
+        self.UseEMpT = UseEMpT
+
+    def open_root(self, path):
+        r_file = ROOT.TFile.Open(path)
+        if not r_file.__nonzero__() or not r_file.IsOpen(): raise NameError('File ' + path + ' not open')
+        return r_file
+
+    def get_root_obj(self, root_file, obj_name):
+        r_obj = root_file.Get(obj_name)
+        if not r_obj.__nonzero__(): raise NameError('Root Object ' + obj_name + ' not found')
+        return r_obj
+
+    def beginJob(self):
+        pass
+
+    def endJob(self):
+        pass
+
+    def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+        self.out = wrappedOutputTree
+        self.branchnames = ["PrefireWeight", "PrefireWeight_Up", "PrefireWeight_Down"]
+        for bname in self.branchnames:
+          self.out.branch(bname, "F")
+
+    def endFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+        pass
+
+    def analyze(self, event):
+        """process event, return True (go to next module) or False (fail, go to next event)"""
+
+        jets = Collection(event,"Jet")
+
+        # Options
+        self.JetMinPt = 20 # Min/Max Values may need to be fixed for new maps
```

Comment:
They used to change a bit whenever the maps were updated, but since the latest maps are pretty much final, these min/max values should be always good.